packages/lib-react-components/src/FavouritesButton/FavouritesButton.spec.js

```diff
@@ -1,100 +1,72 @@
-import { shallow } from 'enzyme'
-import sinon from 'sinon'
-import MetaToolsButton from '../MetaToolsButton'
-
-import { FavouritesButton } from './FavouritesButton'
-import HeartIcon from './HeartIcon'
+import { Grommet } from 'grommet'
+import zooTheme from '@zooniverse/grommet-theme'
+import { render, screen, fireEvent } from '@testing-library/react'
+import { composeStory } from '@storybook/react'
+import Meta, { NotFavourited, Favourited, Disabled } from './FavouritesButton.stories.js'
 
 describe('Component > FavouritesButton', function () {
-  let wrapper
-  const mockTheme = {
-    global: {
-      colors: {
-        statusColors: {
-          error: 'status-error'
-        }
-      }
-    }
-  }
-  before(function () {
-    wrapper = shallow(<FavouritesButton theme={mockTheme} />)
-  })
-
-  it('should render without crashing', function () {
-    expect(wrapper).to.be.ok()
-  })
-
-  it('should display an empty icon', function () {
-    const button = wrapper.find(MetaToolsButton)
-    const { icon } = button.props()
-    expect(icon).to.deep.equal(<HeartIcon color='dark-5' fill='none' size='15px' />)
-  })
-
-  it('should not be checked', function () {
-    const checked = wrapper.prop('aria-checked')
-    expect(checked).to.be.false()
-  })
-
-  describe('on click', function () {
-    let onClickStub
-
-    before(function () {
-      onClickStub = sinon.stub()
-      wrapper = shallow(<FavouritesButton theme={mockTheme} checked={false} onClick={onClickStub} />)
+  describe('when not favourited', function () {
+    beforeEach(function () {
+      const NotFavouritedStory = composeStory(NotFavourited, Meta)
+      render(<Grommet theme={zooTheme}><NotFavouritedStory /></Grommet>)
     })
 
-    afterEach(function () {
-      onClickStub.resetHistory()
+    it('should not show a favourited icon', function () {
+      expect(screen.getByRole('checkbox', { checked: false })).to.exist()
+      expect(screen.getByText('Add to favorites')).to.exist()
     })
 
-    it('should toggle favourites on', function () {
-      wrapper.simulate('click')
-      const icon = wrapper.prop('icon')
-      expect(icon.props.fill).to.equal(mockTheme.global.colors.statusColors.error)
+    it('should toggle favourites on', async function () {
+      fireEvent.click(screen.getByRole('checkbox', { checked: false}))
```

Comment:
`fireEvent` is the old version 13 event testing API. There's a new asynchronous API in version 14, which better simulates real user interactions.
https://testing-library.com/docs/user-event/intro

```js
import userEvent from '@testing-library/user-event'

// inlining
test('trigger some awesome feature when clicking the button', async () => {
  const user = userEvent.setup()
  // Import `render` and `screen` from the framework library of your choice.
  // See https://testing-library.com/docs/dom-testing-library/install#wrappers
  render(<MyComponent />)

  await user.click(screen.getByRole('button', {name: /click me!/i}))

  // ...assertions...
})
```

Reply:
Got these changed